minio/minio
# use rename() safety for in-place 'xl.meta' updates

in the namespace

This makes it safer when replication metadata updates and listing occurs in parallel as listing does not take a lock.

## Community Contribution License
All community contributions in this pull request are licensed to the project maintainers
under the terms of the [Apache 2 license](https://www.apache.org/licenses/LICENSE-2.0). 
By creating this pull request I represent that I have the right to license the 
contributions to the project maintainers under the Apache 2 license.

## Description


## Motivation and Context
Avoid listing missing entries when concurrent replication activity is updating metadata

## How to test this PR?
- adapt this docker-compose https://github.com/minio/openlake/tree/main/spark/partition-test to set up site replication b/w 2 minio nodes 
-  exclude tmp prefixes from versioning -  --excluded-prefixes "*/_temporary/,*/.hive-staging,*/_staging for the bucket
- mark a site offline
- then run the spark notebook with repeated drop, create, insert table sequence
At some point , insert should fail with master branch with location already exists error.

## Types of changes

## Checklist:
